adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -74,16 +77,52 @@ class FastqRecordConverter extends Serializable with Logging {
         .build())
   }
 
+  def convertFragment(element: (Void, Text)): Fragment = {
+    val lines = element._2.toString.split('\n')
+    require(lines.length == 8, "Record has wrong format:\n" + element._2.toString)
```

Comment:
I agree. However, there's a fairly long history to this problem. If you want to track back through it, I might start with PR #679. I think we should fix the FASTQ record parser to support the full FASTQ spec, but it is a low priority from my perspective, which is really just me that no one has come to me/the list screaming about it.